frontend_tests/node_tests/stream_data.js

```diff
@@ -635,3 +641,19 @@ zrequire('marked', 'third/marked/lib/marked');
 
     assert.equal(page_params.notifications_stream, "foo");
 }());
+
+(function test_invite_streams() {
+    // add default stream
+    var orie = {
+        stream_id: 320,
+        name: 'Orie',
+        subscribed: true,
+    };
+
+    stream_data.clear_subscriptions();
+    stream_data.add_sub('Orie', orie);
+    stream_data.set_realm_default_streams([orie]);
+
```

Comment:
You should probably add streams that aren't subscribed and aren't default. You should also add default streams that aren't subscribed.

Reply:
Hey fix both of them.
Just so its clear i add a stream that gets subscribed `Inviter`. 
(I wasn't able to figure out how we subscribe somebody to stream, with `stream_data.add_subscriber`)
And all the default streams are returned so adding new unsubscribed one will not make a difference.

Reply:
I still don't understand why we need "mary" here.

Reply:
If we do not add "mary" to `inviter` stream, `inviter` would not returned from the function `invited_streams`. To be more clear the function returns `_.union` subscribed streams and default realm streams.

https://github.com/zulip/zulip/blob/898acc427820f9ecd1f65e2211070f865d99ec66/static/js/stream_data.js#L134-L138

Reply:
Ok, I think we can make that more explicit by
* add comments explaining what's going on
* do two calls/assertion steps (one before adding her, one after adding her, to show how that changes things)

Reply:
Ok that would be more clear. Doing it soon...
(Sorry for lot of your time)

Reply:
Never mind, we do not need "mary" up here because we just need to add stream with option `subscribed: true`. Updated.